### ISSUE TITLE
java.io.FileNotFoundException with --filter in html5 transform

### ISSUE THREAD

**Issue by lionelmoi** [dita-ot/dita-ot#2593]
I get `FODC0002: java.io.FileNotFoundException:` when setting a ditaval with the --filter option but not with the deprecated `-Ddita.input.valfile`.

It does not seem to affect output but the error is misleading.

In the `--filter`  trace I get these lines:

` [echo] dita.input.valfile=[DITAVAL FOLDER PATH]\\L1_help_publi.ditaval`
` [echo] dita.input.valfile.url=[DITAOT PATH]/$%7Bdita.input.valfile%7D`

`    [xslt] Recoverable error on line 114 of topic.xsl:`
`    [xslt]   FODC0002: java.io.FileNotFoundException:`
`    [xslt]   [DITAOT PATH]\${dita.input.valfile} (Le fichier spécifié est introuvable)`

It seems DITA-OT tries to find the ditaval file in the DITA-OT folder.

**Comment by xephon2** [comment]
Hi,

I don't think, that's a bug. The path should be absolute, please try `${basedir}/path/to/val/dir/L1_help_publi.ditaval`.

Greetings,
Stefan

**Comment by lionelmoi** [comment]
What do you mean by `${basedir}`? the DITA-OT dir ?

**Comment by xephon2** [comment]
`${basedir}` is an Ant property that points to the path of your Ant file, which is usually in the root directory of your DITA project. You need this property to build an absolute path, which is mandatory for DITAVAL references. You can also override the `${basedir}` property by setting the `basedir` attribute of the `<project>` element in your Ant build script, see [Using Apache Ant - Writing a Simple Buildfile](https://ant.apache.org/manual/using.html), but this is probably not what you want. So, if your Ant script is in `/home/user/workspace/dita`, the value of `${basedir}` is `/home/user/workspace/dita`. You do not need to manually set the property, it is already defined by pure **Ant magic**. :boom: You can test this by adding an `<echo>` to your Ant script:

```xml
<target name="foo">
  <echo level="info">basedir=${basedir}</echo>
</target>
```

**Comment by lionelmoi** [comment]
I am giving an absolute path. I really do not think this is the cause. The issue seems to stem from the `dita.input.valfile.url` property that appears to be build incorrectly.

**Comment by xephon2** [comment]
Could you please test this with DITA-OT 2.3.3?

**Comment by lionelmoi** [comment]
The `dita.input.valfile` is initialized by the `build-init` target.

Since the html5 plugin calls the `html5.init` target before the `build-init` target, the property is not initialized. That is why the `dita.input.valfile.url`  `makeurl` task yields an incorrect URL.

I do not know the target are called in this order. If there are no blocking reasons maybe we could reverse the order in the 2.5 release.

@xephon2 I am working with the 2.4.2 version and the double dash arguments. There no such arguments in the 2.3.3. What bugs with `--filter` does not with `-filter`.

I have no idea why but it is troubling.

**Comment by shaneataylor** [comment]
I think this might be related to #2559.

**Comment by lionelmoi** [comment]
@shaneataylor all right it is the same. 

@jelovirt will it be fixed in 2.4.4? I can help out if need be.

**Comment by robander** [comment]
Just found this open issue, but it looks like the duplicate was fixed in 2.4.5 -- @lionelmoi do you know if this issue was resolved by the fix to 2559?

**Comment by lionelmoi** [comment]
You can indeed close this issue.
